### PR TITLE
add vgg16 performance script

### PR DESCRIPTION
**Pull request by CarloLucibello**
We could start collecting a few (possibly more granular) perf scripts, as a first crude benchmarking facility. 

Related discussion in #1527 .

On my laptop,  performances are the same for Flux master and v0.11.6.

```julia
# Julia 1.6-rc1, Flux v0.11.6, CUDA v2.6.1, Zygote v0.6.3, NNlib v0.7.16
CPU benchmark
  forward
  1.752 s (915 allocations: 288.05 MiB)
  backward
  4.166 s (20840 allocations: 1.01 GiB)
  forw and back
  6.117 s (51836 allocations: 1.63 GiB)
CUDA benchmark
  forward
  21.032 ms (4115 allocations: 156.30 KiB)
  backward
  139.990 ms (9218 allocations: 365.88 KiB)
  forw and back
  168.142 ms (19244 allocations: 2.57 MiB)
```
```julia
# Julia 1.6-rc1, Flux master, CUDA v2.6.1, Zygote v0.6.3, NNlib v0.7.16
CPU benchmark
  forward
  1.682 s (1123 allocations: 220.57 MiB)
  backward
  4.210 s (20153 allocations: 1.08 GiB)
  forw and back
  6.382 s (32100 allocations: 1.69 GiB)
CUDA benchmark
  forward
  21.704 ms (4052 allocations: 149.59 KiB)
  backward
  139.710 ms (9832 allocations: 377.77 KiB)
  forw and back
  169.053 ms (20759 allocations: 2.32 MiB)
```
```julia

```

### CODE REVIEW

**Comment by darsnack** [comment]
This is great! I think the SciML benchmarks reside in a separate repo. I'm not sure if this is a subjective choice or a technical one (@ChrisRackauckas or @DhairyaLGandhi would can comment). Ideally, `ModelZooKeeper` will run these in the future, but even having them in the repo right now will make us all feel better about whether a release is ready.

One change I would make is to structure the directory like:
```plaintext
benchmarks
\_ metalhead_vgg16
   \_ Manifest.toml
   \_ Project.toml
   \_ script.jl
...
\_ run_all.jl
```
Each benchmark gets its own sub-directory with its own environment to be exactly reproducible on everyone's machines (and CI).

If we get this merged in, I can submit another PR with more vision related benchmarks.

**Comment by ChrisRackauckas** [comment]
It's because the benchmark results can get huge. https://github.com/SciML/SciMLBenchmarks.jl hosts a huge set of PDFs and webpages to display the current benchmarks. You don't want to ship all of that with every install

**Comment by DhairyaLGandhi** [comment]
I have some work on benchmarks that I have been coordinating already. We will populate a public https://speed.fluxml.ai soon

**Comment by DhairyaLGandhi** [comment]
Also yes, done automatically as a cron job, and not part of this repo.

**Comment by darsnack** [comment]
Nice, is it possible for us to PR/commit to whatever feeds that website? I'm just thinking to the original motivation for this PR — having one set of benchmarks that we are all referring to as we track whether code is ready to be released.

**Comment by CarloLucibello** [comment]
> I have some work on benchmarks that I have been coordinating already. We will populate a public https://speed.fluxml.ai soon

That seems cool.

So let's merge this as a short term solution, then we can expand, move somewhere else, ditch entirely when a better solution becomes available.

> One change I would make is to structure the directory like:

I added some other scripts (included a runall) and Project/Manifest, but kept a flat structure. Feel free to modify in the future.

**Comment by CarloLucibello** [comment]
merge @darsnack ?

File: perf/runbenchmarks.jl [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using Flux
+
```
Comment (darsnack) [inline perf/runbenchmarks.jl:2]: Print `versioninfo()`?

Comment (CarloLucibello) [inline perf/runbenchmarks.jl:2]:
```suggestion

versioninfo()
```

Comment (CarloLucibello) [reply perf/runbenchmarks.jl:2]: done

**Review by darsnack — APPROVED** [review]
Good by me. @DhairyaLGandhi are the benchmarks you mentioned usable _now_? If so, we don't need to merge this. If not, let's merge this just so we can track in preparation for the upcoming release. We can transition to a more complete solution in a separate repo later.

**Comment by CarloLucibello** [comment]
merging. No big deal removing if not needed

bors r+

**Pull request merged**